youtube-client-app/src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { Router } from '@angular/router';
+import { IUser } from '../../shared/models/user.models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  public user$ = new Subject<IUser | null>();
+
+  private user!: IUser | null;
```

Comment:
it is better to keep state in one place (usually in observable), so you don't have to deal with it twice like in these examples:
```ts
    this.user = user;
    this.user$.next(user);
```
```ts
    this.user$.next(null);
    this.user = null;
```
you can read more about why state duplication is bad in the internet)